### PR TITLE
Create accordion.html.twig

### PR DESCRIPTION
**Pull request by wucherpfennig**
## Description
Add accordion component. see #99 

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist
- [x] I updated the documentation (see [here](https://github.com/kevinpapst/TablerBundle/tree/master/docs))
- [x] I agree that this code will be published under the [MIT license](https://github.com/kevinpapst/TablerBundle/blob/master/LICENSE)

### CODE REVIEW

**Comment by wucherpfennig** [comment]
I did not expect this to happen... ;-) 

I assume this is because of the added lock files from the last commit. 
I think it is up to you to either "fix" the lock file or you decide to continue with php > 7.4 @kevinpapst 

same goes for the other PR I suggested #104

**Comment by kevinpapst** [comment]
Nah. My mistake. I committed composer.lock which was created with PHP 8.1 - I'll check that tomorrow

**Comment by kevinpapst** [comment]
Sorry for that, can you please merge master into your branches

**Comment by wucherpfennig** [comment]
Done

**Comment by kevinpapst** [comment]
And one last sorry for forgetting to come back to this PR.
What about a short docs page and I will add it then to the example app?

**Comment by wucherpfennig** [comment]
Sorry for the long delay. I have added the required docs.

**Comment by wucherpfennig** [comment]
Updated docs for this PR too.

**Comment by cavasinf** [comment]
I was about to do it, need that component 👍

File: docs/components-accordion.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+See Tabler documentation at https://preview.tabler.io/accordion.html
+
+## Accordion
+
+### Parameters
+`accordion()` macro, waits for 2 parameters:
+
+| Parameter | Description                |   Type    | Default |
+|:---------:|:---------------------------|:---------:|:-------:|
+|   items   | Array of [Item](#Step)     |  `array`  |  `[]`   |
```
Comment (kevinpapst) [inline docs/components-accordion.md:14]:
```suggestion
|   items   | Array of [Item](#Item)     |  `array`  |  `[]`   |
```

File: docs/components-accordion.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+          {
+            'extraClass': 'bg-danger-lt'
+          }
+        },
+        'key4':
+        {
+          'title': '<div>Title 4 <strong>with HTML</strong></div>',
+          'body': '<div>Body 4 <strong>with HTML</strong></div',
+        },
+    } %}
```
Comment (kevinpapst) [inline docs/components-accordion.md:71]: The docs above say array of objects, here we have an object of objects. Lets change that, if the key is not required
```suggestion
{% set items = [
        {
          'title': 'Title 1',
          'body': 'Body 1',
        },
        {
          'title': 'Title 1',
          'body': 'Body 2',
          'options':
          {
            'title_extraClass': 'bg-success text-white',
            'body_extraClass': 'bg-danger pt-3',
          },
        },
        {
          'title': 'Title 3',
          'body': 'Body 3',
          'options':
          {
            'extraClass': 'bg-danger-lt'
          }
        },
        {
          'title': '<div>Title 4 <strong>with HTML</strong></div>',
          'body': '<div>Body 4 <strong>with HTML</strong></div',
        },
    ] %}
```

File: templates/components/accordion.html.twig [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    {% set _always_open = options.always_open ?? false %}
+    {% set _extraClass = options.extraClass ?? '' %}
+
+    <div id="{{ _id }}" class="accordion {% if _flush == true %}accordion-flush{% endif %} {{ _extraClass }}">
+        {% for item in items %}
+            {% set _id_collapse = _id ~ '-collapse-' ~ loop.index %}
+            {% set _item_extraClass = item.options.extraClass ?? '' %}
+            {% set _item_title_extraClass = item.options.title_extraClass ?? '' %}
+            {% set _item_body_extraClass = item.options.body_extraClass ?? '' %}
+            <div class="accordion-item {{ _item_extraClass }}">
```
Comment (kevinpapst) [inline templates/components/accordion.html.twig:14]: Why are those called `xxx_extraClass`? Wouldn't be `xxx_class` sufficient?

**Review by kevinpapst — CHANGES_REQUESTED** [review]

**Comment by kevinpapst** [comment]
@cavasinf the best review is using the component, could you give it a try already?

File: templates/components/accordion.html.twig [hunk 4]
```diff
@@ -0,0 +1,31 @@
+{% macro accordion(items, options) %}
+    {% set _id = options.id ?? ('accordion-' ~ random(0, 999)) %}
+    {% set _raw = options.raw ?? true %}
+    {% set _flush = options.flush ?? false %}
+    {% set _always_open = options.always_open ?? false %}
```
Comment (cavasinf) [inline templates/components/accordion.html.twig:5]: As we said before @kevinpapst, we should keep the code to `CamelCase` only.
`always_open` ➡️ `alwaysOpen `

User macro's options should always be like that, but OUR code can be like `title_class` or `titleClass` ?

File: templates/components/accordion.html.twig [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+            {% set _id_collapse = _id ~ '-collapse-' ~ loop.index %}
+            {% set _item_extraClass = item.options.extraClass ?? '' %}
+            {% set _item_title_extraClass = item.options.title_extraClass ?? '' %}
+            {% set _item_body_extraClass = item.options.body_extraClass ?? '' %}
+            <div class="accordion-item {{ _item_extraClass }}">
+                <div class="accordion-header {{ _item_title_extraClass }}" id="{{ _id }}-heading-{{ loop.index }}">
+                    <button class="accordion-button {% if loop.first == false %}collapsed{% endif %}"
+                            type="button" data-bs-toggle="collapse" data-bs-target="#{{ _id_collapse }}"
+                            aria-expanded="true">
+                        {% if _raw == true %}{{ item.title | raw }}{% else %}{{ item.title }}{% endif %}
```
Comment (cavasinf) [inline templates/components/accordion.html.twig:19]:
```suggestion
                        {% if _raw %}{{ item.title | raw }}{% else %}{{ item.title }}{% endif %}
```

File: templates/components/accordion.html.twig [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+                    </button>
+                </div>
+                <div id="{{ _id_collapse }}" {% if _always_open == false %}data-bs-parent="#{{ _id }}"{% endif %}
+                     class="accordion-collapse collapse {% if loop.first %}show{% endif %} {{ _item_body_extraClass }}">
+                    <div class="accordion-body pt-0">
+                        {% if _raw == true %}{{ item.body | raw }}{% else %}{{ item.body }}{% endif %}
```
Comment (cavasinf) [inline templates/components/accordion.html.twig:25]:
```suggestion
                        {% if _raw %}{{ item.body | raw }}{% else %}{{ item.body }}{% endif %}
```

File: templates/components/accordion.html.twig (hunk 5 above)
Comment (cavasinf) [inline templates/components/accordion.html.twig:19]: @wucherpfennig what's happening if there is no `title` with the item ?

File: docs/components-accordion.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    ] %}
+
+{% set options = {
+    'id': 'example-accordion',
+    'raw': null,
+    'always_open': false,
+    'extraClass': 'mb-0',
+} %}
+
+{{ accordion(items, options) }}
```
Comment (cavasinf) [inline docs/components-accordion.md:75]:
```suggestion
### Parameters
`accordion()` macro, waits for 2 parameters:

| Parameter | Description                |   Type    | Default |
|:---------:|:---------------------------|:---------:|:-------:|
|   items   | Array of [Item](#Item)     |  `array`  |  `[]`   |
|  options  | [Options](#Options) object | `object`  |  `{}`   |

#### Item
| Parameter | Description                                          |   Type   |    Default     |
|:---------:|------------------------------------------------------|:--------:|:--------------:|
|   title   | Title of the item                                    | `string` |      `#`       |
|   body    | Content of the item, can be `HTML` with `raw` option | `string` | _empty string_ |
|  options  | Customize the [Item Options](#Item Options)          | `object` |      `{}`      |

##### Item Options
|    Parameter     | Description              |   Type   |    Default     |
|:----------------:|--------------------------|:--------:|:--------------:|
| title_extraClass | Class for the item title | `string` | _empty string_ |
| body_extraClass  | Class for the item body  | `string` | _empty string_ |

#### Options
|  Parameter  | Description                              |   Type    |             Default             |
|:-----------:|------------------------------------------|:---------:|:-------------------------------:|
|     id      | Set custom `id` to accordion             | `string`  | `'accordion-'` + `rand(0, 999)` |                                        
|     raw     | Use raw output for `title` and `body`    | `boolean` |             `true`              |                                        
|    flush    | Remove padding. See tabler.              | `boolean` |             `false`             | 
| always_open | Do not close previous items on click.    | `boolean` |             `false`             |   
| extraClass  | Add extra classes on accordion container | `string`  |         _empty string_          |      

### Usage

```twig
{% from '@Tabler/components/accordion.html.twig' import accordion %}

{% set items = [
    {
        'title': 'Title 1',
        'body': 'Body 1',
    },
    {
        'title': 'Title 2',
        'body': 'Body 2',
        'options':
        {
            'title_extraClass': 'bg-success text-white',
            'body_extraClass': 'bg-danger pt-3',
        },
    },
    {
        'title': 'Title 3',
        'body': 'Body 3',
        'options':
        {
            'extraClass': 'bg-danger-lt'
        }
    },
    {
        'title': '<div>Title 4 <strong>with HTML</strong></div>',
        'body': '<div>Body 4 <strong>with HTML</strong></div',
    },
] %}

{% set options = {
    'id': 'example-accordion',
    'raw': true,
    'always_open': false,
    'extraClass': 'mb-0',
} %}

{{ accordion(items, options) }}
```

Comment (cavasinf) [reply docs/components-accordion.md:75]: I'm not OK with `flush` option description.
There's no information about that in Tabler, but can be found with Bootstrap https://getbootstrap.com/docs/5.0/components/accordion/#flush

File: templates/components/accordion.html.twig [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+    <div id="{{ _id }}" class="accordion {% if _flush == true %}accordion-flush{% endif %} {{ _extraClass }}">
+        {% for item in items %}
+            {% set _id_collapse = _id ~ '-collapse-' ~ loop.index %}
+            {% set _item_extraClass = item.options.extraClass ?? '' %}
+            {% set _item_title_extraClass = item.options.title_extraClass ?? '' %}
+            {% set _item_body_extraClass = item.options.body_extraClass ?? '' %}
+            <div class="accordion-item {{ _item_extraClass }}">
+                <div class="accordion-header {{ _item_title_extraClass }}" id="{{ _id }}-heading-{{ loop.index }}">
+                    <button class="accordion-button {% if loop.first == false %}collapsed{% endif %}"
```
Comment (cavasinf) [inline templates/components/accordion.html.twig:16]: `is item open` should be an item option, not if not first loop.

Example, I want to have the second accordion item to be opened when arriving in the page.

**Review by cavasinf — COMMENTED** [review]
Tweaking+ Reformatting the doc

**Comment by cavasinf** [comment]
@kevinpapst I'm in holidays RN, only thing I can do is "check by eye" and test it next Monday.

**Comment by cavasinf** [comment]
I've already suggested a DOC update, but it is outdated now 😢 
But it was almost like yours now 👍

**Comment by kevinpapst** [comment]
I've seen that (thanks 😄 ) - but it was impossible to read, as it was so long and Github rendered it awfully, no chance to understand changes.

**Comment by wucherpfennig** [comment]
Hey guys what do I need to do to help you? (Writing better code in the first place is not an option 😉)

**Comment by kevinpapst** [comment]
@wucherpfennig you could update to use my latest changes and then check the comments by @cavasinf 

I haven't done a demo implementation yet, will do and provide feedback as well asap.

**Review by wucherpfennig — COMMENTED** [review]
looks good

**Comment by wucherpfennig** [comment]
Oh come on... tbh I do not know what else I should do ;-)

File: docs/components-accordion.md [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+|:---------------:|------------------------------|:--------:|:--------------:|
+| titleExtraClass | Add extra class to the title | `string` | _empty string_ |
+| bodyExtraClass  | Add extra class to the body  | `string` | _empty string_ |
+
+#### Options
+|  Parameter  | Description                              |   Type    |    Default     |
+|:-----------:|------------------------------------------|:---------:|:--------------:|
+|     id      | Set custom `id` to accordion             |  `mixed`  | `rand(number)` |                                        
+|     raw     | Use raw output for `title` and `body`    | `boolean` |     `true`     |                                        
+|    flush    | Remove padding (see Tabler docs)         | `boolean` |    `false`     |
```
Comment (cavasinf) [inline docs/components-accordion.md:35]: I'm not OK with flush option description.
There's no information about that in Tabler, but can be found with Bootstrap https://getbootstrap.com/docs/5.0/components/accordion/#flush

**Comment by cavasinf** [comment]
LGTM now 👍

**Review by kevinpapst — APPROVED** [review]
I'll merge, add a demo page and check if everything works as expected.
Thanks @wucherpfennig 😄

**Pull request merged**